### DISCUSSION TITLE
Container queries support for native

### DISCUSSION THREAD

**Discussion by avedetvedea** [nativewind/nativewind#341] [General]
I think container queries suit native design even more than layout changes with media queries. Looking forward to having the equivalent of tw 3.2 container queries in natewind.

**Comment by marklawlor** [comment]
I've been thinking about this as well. I believe NativeWind can support a subset of the container query spec, but there will be some implementation differences.

For example, NativeWind cannot measure the size of components until they have been rendered - so we might have to render the children with `opacity: 0`, measure the parent, perform the style updates and then set `opacity: 1`. Hopefully this would avoid the flash of incorrectly styled components. Dripsy takes a similar approach with their implementation https://github.com/nandorojo/dripsy/blob/74c6fcfb0dfc40a709743db1bba8af209e3eaea1/packages/core/src/container-query/index.tsx

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
